Proton/Sources/Core/PRTextStorage.m

```diff
@@ -153,71 +133,24 @@ - (void)setAttributes:(NSDictionary<NSString *,id> *)attrs range:(NSRange)range
 }
 
-- (NSDictionary<NSAttributedStringKey,id> *)applyingDefaultFormattingIfRequiredToAttributes:(NSDictionary<NSAttributedStringKey,id> *)attributes {
-    NSMutableDictionary<NSAttributedStringKey,id> * updatedAttributes = attributes.mutableCopy;
-    if (!updatedAttributes) {
-        updatedAttributes = [[NSMutableDictionary alloc] init];
-    }
-
-    if (![attributes objectForKey:NSParagraphStyleAttributeName]) {
-        id value = _defaultTextFormattingProvider.paragraphStyle;
-        if (!value) {
-            value = [self defaultParagraphStyle];
-        }
-        [updatedAttributes setObject:value forKey:NSParagraphStyleAttributeName];
-    }
-
-    if (![attributes objectForKey:NSFontAttributeName]) {
-        id value = _defaultTextFormattingProvider.font;
-        if (!value) {
-            value = [self defaultFont];
-        }
-        [updatedAttributes setObject:value forKey:NSFontAttributeName];
-    }
-
-    if (![attributes objectForKey:NSForegroundColorAttributeName]) {
-        id value = _defaultTextFormattingProvider.textColor;
-        if (!value) {
-            value = [self defaultTextColor];
-        }
-        [updatedAttributes setObject:value forKey:NSForegroundColorAttributeName];
-    }
-
-    return updatedAttributes;
-}
-
-- (void)fixMissingAttributesForDeletedAttributes:(NSArray<NSAttributedStringKey> *)attrs range:(NSRange)range {
-    if ([attrs containsObject:NSForegroundColorAttributeName]) {
-        [_storage addAttribute:NSForegroundColorAttributeName value:[self defaultTextColor] range:range];
-    }
-
-    if ([attrs containsObject:NSParagraphStyleAttributeName]) {
-        [_storage addAttribute:NSParagraphStyleAttributeName value:[self defaultParagraphStyle] range:range];
-    }
-
-    if ([attrs containsObject:NSFontAttributeName]) {
-        [_storage addAttribute:NSFontAttributeName value:[self defaultFont] range:range];
-    }
-}
-
 - (void)insertAttachmentInRange:(NSRange)range attachment:(Attachment *_Nonnull)attachment {
-    id spacer = attachment.spacer.string;
-    bool hasPrevSpacer = false;
+    NSString *spacer = attachment.spacer.string;
+    BOOL hasPrevSpacer = NO;
     if (range.length + range.location > 0) {
         NSRange subrange = NSMakeRange(range.location == 0 ? 0 : range.location - 1, 1);
-        hasPrevSpacer = [self attributedSubstringFromRange: subrange].string == spacer;
+        hasPrevSpacer = [self attributedSubstringFromRange:subrange].string == spacer;
     }
-    bool hasNextSpacer = false;
+    BOOL hasNextSpacer = NO;
     if ((range.location + range.length + 1) <= self.length) {
         NSRange subrange = NSMakeRange(range.location, 1);
-        hasNextSpacer = [self attributedSubstringFromRange: subrange].string == spacer;
+        hasNextSpacer = [self attributedSubstringFromRange:subrange].string == spacer;
     }
 
-    NSAttributedString *attachmentString = [attachment stringWithSpacersWithAppendPrev:!hasNextSpacer appendNext:!hasNextSpacer];
```

Comment:
`hasPrevSpacer` wasn't used before. However, there is no difference because in this `stringWithSpacers` method this parameter is not used
<img width="881" alt="image" src="https://user-images.githubusercontent.com/53595/93730253-314a1580-fc0b-11ea-972f-615ab7e3f7f3.png">
@rajdeep why this code is disabled?

Reply:
This PR is good progress. I'd just like closure on what is happening here.

Reply:
Not able to recall right now, but as far as I remember, I was running into some issues. The idea was if you would like to have a space before and after the attachment. At that time, it seemed like an overkill owing to the issues I ran into, so commented it out to have a look at it again sometime in future.